lib/Conversion/FIRRTLToRTL/LowerToRTL.cpp

```diff
@@ -104,6 +106,78 @@ static bool isZeroBitFIRRTLType(Type type) {
   return type.cast<FIRRTLType>().getPassiveType().getBitWidthOrSentinel() == 0;
 }
 
+namespace {
+struct FirMemory {
+  size_t numReadPorts;
+  size_t numWritePorts;
+  size_t numReadWritePorts;
+  size_t dataWidth;
+  size_t depth;
+  size_t readLatency;
+  size_t writeLatency;
+  bool operator<(const FirMemory &rhs) const {
+#define cmp3way(name)                                                          \
+  if (name < rhs.name)                                                         \
+    return true;                                                               \
+  if (name > rhs.name)                                                         \
+    return false;
+    cmp3way(numReadPorts);
+    cmp3way(numWritePorts);
+    cmp3way(numReadWritePorts);
+    cmp3way(dataWidth);
+    cmp3way(depth);
+    cmp3way(readLatency);
+    cmp3way(writeLatency);
+    return false;
+#undef cmp3way
+  }
+};
+} // namespace
+
+static std::string getFirMemoryName(const FirMemory &mem) {
+  return llvm::formatv("FIRRTLMem_{0}_{1}_{2}_{3}_{4}_{5}_{6}",
+                       mem.numReadPorts, mem.numWritePorts,
+                       mem.numReadWritePorts, mem.dataWidth, mem.depth,
+                       mem.readLatency, mem.writeLatency);
```

Comment:
Does it make sense to add some indicator of what these are, e.g.:

```
"FIRRTLMem_r{0}w{1}rw{2}_{3}x{4}_rlat{5}_wlat{6}"
```

Reply:
This is intended as a unique symbol name, not a meaning human readable name.